librecad/src/lib/actions/rs_snapper.cpp

```diff
@@ -758,22 +771,56 @@ void RS_Snapper::drawSnapper()
                 }
                 else //orthogonal crosshair
                 {
+					double snapX=graphicView->toGuiX(pImpData->snapCoord.x);
+					double snapY=graphicView->toGuiY(pImpData->snapCoord.y);
+
+					double viewWidth=double(graphicView->getWidth());
+					double viewHeight=double(graphicView->getHeight());
+
 					if (pImpData->snap_indicator=="Crosshair")
                     {
-                        RS_OverlayLine *line=new RS_OverlayLine(nullptr,
-						{{0., graphicView->toGuiY(pImpData->snapCoord.y)},
-						 {double(graphicView->getWidth()),
-						  graphicView->toGuiY(pImpData->snapCoord.y)}
-                                                                });
-						line->setPen(pImpData->line_pen);
-                        container->addEntity(line);
+                        RS_OverlayLine *line;
+
+						// ----O     (Left)
+						line=new RS_OverlayLine(nullptr, {
```

Comment:
@Osndok @r-a-v-a-s 

sorry, my bad.

at line 710 of RS_Snapper, the "Snapper" container is cleared

```C++
    graphicView->getOverlayContainer(RS2::Snapper)->clear();
```
and this container is created by default autoDelete=true. Therefore, there is no memory leak.